src/Configuration.cpp

```diff
@@ -107,4 +108,27 @@ std::string Configuration::getClientId()  const {
 
 std::string Configuration::getClientSecret()  const {
     return clientSecret;
+}
+
+void Configuration::validate_configuration_values() {
+
+    if(sampleRate != 8000 and sampleRate != 16000) {
+        throw std::runtime_error("Unsupported sample rate value. Allowed values: 8000 or 1600");
+    }
+
+    if (std::find(allowedTopicValues.begin(), allowedTopicValues.end(), topic) == allowedTopicValues.end())
+    {
+        throw std::runtime_error("Unsupported topic value. Allowed values: GENERIC, BANKING, TELCO, INSURANCE");
+    }
+    
+    if (std::find(allowedLanguageValues.begin(), allowedLanguageValues.end(), language) == allowedLanguageValues.end())
+    {
+        throw std::runtime_error("Unsupported language value. Allowed values: en-US, en-GB, pt-BR, es, es-419, tr, ja, fr, fr-CA, de, it");
```

Comment:
Maybe it would be best to use the allowedValues vector to automatically fill this string so we do not make a mistake when we add a new language